### DISCUSSION TITLE
RotateView issue on Android v13

### DISCUSSION THREAD

**Discussion by karel-zabloudil-3pg** [mapsforge/mapsforge#1452] [Q&A] [answered]
Hi,

We are currently facing an issue with RotateView. When the view is rotated then the map dissapears.
I also tried the official example (RotateMapViewer button) in the mapsforge repository with the same result. 

I tried mapsforge v0.20.0 as well as the latest version in master with the same result. 
Interestingly this issue does not exists on Android v10.

Has anybody experienced same issue ?

thanks,
Karel

**Comment by devemux86** [comment, marked as answer]
See the VTM (OpenGL) map library, compatible with Mapsforge maps: https://github.com/mapsforge/vtm
It has map rotation / tilt, continuous zoom and many other features.

**Reply by karel-zabloudil-3pg** [reply]
Thank you!
Is VTM replacement for "original" mapsforge ?

**Reply by devemux86** [reply]
>  Is VTM replacement for "original" mapsforge ?

Both libraries are maintained.

It would be nice to also add map rotation and continuous zoom to Mapsforge.
We need the community's help for this! :slightly_smiling_face:

**Reply by karel-zabloudil-3pg** [reply]
Just currious, why are the both libraries maintained if they are compatible and from my perspective VTM library performs better than mapsforge ?

**Reply by devemux86** [reply]
Mapsforge also includes the map writer and the [POI api](https://github.com/mapsforge/mapsforge/blob/master/docs/POI.md).

There wasn't much interest from anyone to work on map rotation and continuous zoom.
So we continue development of VTM that is a similar implementation using OpenGL.

**Reply by karel-zabloudil-3pg** [reply]
Also rotation of the mapsforge map is working (through RotateView) in Android 10, but it is not working in Android 13 (map disappears with the blank white view). I also tried the rotation example in official mapsforge example app with the same result.
Should I open Issue ticket ? Thanks.

**Reply by devemux86** [reply]
This is not a proper vector map rotation, e.g. labels must remain without rotation.
It was just an experiment from the old Android days and only for bitmap maps.

You can open an issue if you want, but it was never a recommended working method.
